### DISCUSSION TITLE
How to build and run "ModelTests" unit tests

### DISCUSSION THREAD

**Discussion by georgen117** [microsoft/onnxruntime#7660] [Other Q&A] [answered]
I have one of my pull requests that is failing due to the "ModelTests" unit tests in the CI build.  The failure was really clear and easy to resolve.  However, I have not been able to figure out how to run the "ModelTests" as part of a local build.

I have found the code located in `test\providers\cpu\model_tests.cc.`

If I had seen the failures locally I would not have wasted CI build time.  I would like to know how to build and run these tests locally so I can run them against future commits.

Is there a build flag that I am unaware of that will cause the "ModelTests" to build when onnxruntime_test_all is built?

Is this set of tests only limited to Linux OS or can I also build and run them on Windows?

**Comment by ytaous** [comment, marked as answer]
To run it locally, you need to modify the .cc a little bit and enable "--enable_onnx_tests" in the build command. However, since the input data is not in public, so you won't be able to run it locally.
Example:
./build.sh --config Debug --use_cuda --enable_training --build_wheel --skip_tests --parallel 8 --enable_onnx_tests
./onnxruntime_test_all --gtest_filter=ModelTests*

In fact, the model input is proprietary and private, so we won't be able to share it.

**Reply by georgen117** [reply]
> In fact, the model input is proprietary and private, so we won't be able to share it.

That is unfortunate but understandable.  Thanks for the reply @ytaous. I guess I will have to rely on the CI build system.  At least the error was clear and I was able to resolve the issue without having access to the tests. :)
